File: collect_app/src/androidTest/java/org/odk/collect/android/support/NotificationDrawerRule.java
package org.odk.collect.android.support;

import org.junit.rules.TestRule;
import org.junit.runner.Description;
import org.junit.runners.model.Statement;
import org.odk.collect.android.support.pages.NotificationDrawer;

public class NotificationDrawerRule implements TestRule {

    private final NotificationDrawer notificationDrawer = new NotificationDrawer();
    
    @Override
    public Statement apply(Statement base, Description description) {
        return new Statement() {
            @Override
            public void evaluate() throws Throwable {
                try {
                    base.evaluate();
                } finally {
                    notificationDrawer.teardown();
                }
            }
        };
    }

    public NotificationDrawer open() {
        return notificationDrawer.open();
    }
}

File: app/src/androidTest/java/org/odk/collect/android/support/rules/NotificationDrawerRule.kt
package org.odk.collect.android.support.rules

import org.junit.rules.TestRule
import org.junit.runner.Description
import org.junit.runners.model.Statement
import org.odk.collect.android.support.pages.NotificationDrawer

class NotificationDrawerRule : TestRule {
    private val notificationDrawer = NotificationDrawer()

    override fun apply(base: Statement, description: Description): Statement {
        return object : Statement() {
            @Throws(Throwable::class)
            override fun evaluate() {
                try {
                    base.evaluate()
                } finally {
                    notificationDrawer.teardown()
                }
            }
        }
    }

    fun open(): NotificationDrawer {
        return notificationDrawer.open()
    }
}
